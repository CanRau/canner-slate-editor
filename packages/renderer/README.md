

- [Renderers - @canner/slate-editor-renderer](./packages/slate-editor-renderer)
- [Serializer and deserializer - @canner/slate-editor-html](./packages/slate-editor-html)
