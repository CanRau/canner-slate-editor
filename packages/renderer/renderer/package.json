{
  "name": "@canner/slate-editor-renderer",
  "version": "1.3.0",
  "description": "Renderers for slate nodes and marks",
  "main": "lib/index.js",
  "scripts": {
    "test": "jest src",
    "prepublish": "../../../node_modules/.bin/babel src -d lib --copy-files"
  },
  "author": "",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "devDependencies": {
    "jest": "^22.4.2",
    "react-test-renderer": "^16.2.0"
  },
  "peerDependencies": {
    "react": "^0.14.0 || ^15.0.0 || 16.x",
    "react-dom": "^0.14.0 || ^15.0.0 || 16.x",
    "slate": "^0.33.0",
    "styled-components": "^3.2.1"
  },
  "dependencies": {
    "@canner/slate-helper-inline-adddata": "^0.1.2",
    "emoji-mart": "^2.5.0",
    "js-video-url-parser": "^0.2.7",
    "lodash.mapvalues": "^4.6.0",
    "quill-icons": "^0.6.0",
    "react-icons": "^2.2.7",
    "react-loading-image": "^0.4.2",
    "react-resizable": "^1.7.5",
    "slate-edit-table": "^0.15.0-alpha.4"
  }
}
