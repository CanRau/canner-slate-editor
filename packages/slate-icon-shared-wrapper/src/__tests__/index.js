import React from 'react';
import SharedWrapper from '../';
import renderer from 'react-test-renderer';

test('create a AlignJustify icon', () => {
  const component = renderer.create(
    <SharedWrapper
      type="align-justify"
      icon="AlignJustify"/>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
