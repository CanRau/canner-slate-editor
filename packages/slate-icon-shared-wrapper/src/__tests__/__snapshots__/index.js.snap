// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create a AlignJustify icon 1`] = `
<span
  data-active={false}
  onClick={undefined}
  onMouseDown={[Function]}
>
  <svg
    className={undefined}
    colorLabelClassName={undefined}
    colorStyle={undefined}
    evenClassName={undefined}
    fillClassName={undefined}
    height="18px"
    strokeMitterClassName={undefined}
    thinClassName={undefined}
    transparentClassName={undefined}
    viewBox="0 0 18 18"
    width="18px"
  >
    <line
      className="ql-stroke"
      x1="15"
      x2="3"
      y1="9"
      y2="9"
    />
    <line
      className="ql-stroke"
      x1="15"
      x2="3"
      y1="14"
      y2="14"
    />
    <line
      className="ql-stroke"
      x1="15"
      x2="3"
      y1="4"
      y2="4"
    />
  </svg>
</span>
`;
